vowpalwabbit/core/include/vw/core/learner.h

```diff
@@ -473,25 +473,43 @@ class learner
     _finish_example_fd.print_example_f(all, _finish_example_fd.data, (void*)&ec);
   }
 
-  inline void NO_SANITIZE_UNDEFINED update_stats(VW::workspace& all, const E& ec)
+  inline void NO_SANITIZE_UNDEFINED update_stats(
```

Comment:
These are public learner functions. `data` should not be exposed as a parameter for the caller.